### DISCUSSION TITLE
Local testing

### DISCUSSION THREAD

**Discussion by seddik11** [zkSync-Community-Hub/zksync-developers#37] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

testing in a local environment

### What is the specific issue or error you're encountering?

the bash script for running docker instances is not working

### Can you share the error messages or logs you're receiving, if any?

`Starting local-setup_geth_1     ... done
Starting local-setup_postgres_1 ... done
Starting local-setup_zksync_1   ... done
Attaching to local-setup_zksync_1
zksync_1    | 2023-06-26T15:22:31.941431Z  INFO zksync_server: No sentry url configured
zksync_1    | 2023-06-26T15:22:32.218231Z DEBUG zksync_core::genesis: genesis is not needed!
zksync_1    | 2023-06-26T15:22:32.235711Z  INFO zksync_core: Starting the components: [HttpApi, WsApi, ExplorerApi, Tree, TreeLightweight, EthWatcher, EthTxAggregator, EthTxManager, DataFetcher, StateKeeper, WitnessGenerator(None, BasicCircuits), WitnessGenerator(None, LeafAggregation), WitnessGenerator(None, NodeAggregation), WitnessGenerator(None, Scheduler), Housekeeper]
zksync_1    | 2023-06-26T15:22:32.359970Z  INFO zksync_eth_client::clients::http::signing: Operator address: 0xde03a0b5963f75f1c8485b355ff6d30f3093bde7
zksync_1    | 2023-06-26T15:22:32.456976Z  INFO zksync_eth_client::clients::http::signing: Operator address: 0xde03a0b5963f75f1c8485b355ff6d30f3093bde7
zksync_1    | thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Contract(Api(Transport(Message("failed to send request: error sending request for url (http://geth:8545/): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution"))))', /usr/src/zksync/core/lib/circuit_breaker/src/vks.rs:99:10
zksync_1    | stack backtrace:
zksync_1    |    0:     0x55fa26990c9a - std::backtrace_rs::backtrace::libunwind::trace::h79937bc171ada62c
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
zksync_1    |    1:     0x55fa26990c9a - std::backtrace_rs::backtrace::trace_unsynchronized::h2292bca8571cb919
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
zksync_1    |    2:     0x55fa26990c9a - std::sys_common::backtrace::_print_fmt::h9c461f248e4ae90d
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:65:5
zksync_1    |    3:     0x55fa26990c9a - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::he9fe6bf1a39182e1
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:44:22
zksync_1    |    4:     0x55fa269b712e - core::fmt::write::h032658c119c720d7
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/fmt/mod.rs:1208:17
zksync_1    |    5:     0x55fa2698a985 - std::io::Write::write_fmt::h299fc90dfae41c0d
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/io/mod.rs:1682:15
zksync_1    |    6:     0x55fa26990a65 - std::sys_common::backtrace::_print::heb70d25df9937e3f
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:47:5
zksync_1    |    7:     0x55fa26990a65 - std::sys_common::backtrace::print::had745c0a76b8b521
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:34:9
zksync_1    |    8:     0x55fa269926bf - std::panicking::default_hook::{{closure}}::h1ea782cdfa2fd097
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:267:22
zksync_1    |    9:     0x55fa269923fb - std::panicking::default_hook::h1cc3af63455a163c
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:286:9
zksync_1    |   10:     0x55fa26992dcc - std::panicking::rust_panic_with_hook::h5cafdc4b3bfd5528
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:688:13
zksync_1    |   11:     0x55fa26992b69 - std::panicking::begin_panic_handler::{{closure}}::hf31c60f40775892c
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:579:13
zksync_1    |   12:     0x55fa2699114c - std::sys_common::backtrace::__rust_end_short_backtrace::h28a5c7be595826cd
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:137:18
zksync_1    |   13:     0x55fa26992872 - rust_begin_unwind
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:575:5
zksync_1    |   14:     0x55fa24c43fb3 - core::panicking::panic_fmt::h8fa27a0b37dd98b7
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/panicking.rs:64:14
zksync_1    |   15:     0x55fa24c44443 - core::result::unwrap_failed::h62abf61e411aaa08
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/result.rs:1791:5
zksync_1    |   16:     0x55fa24f031fd - <zksync_circuit_breaker::vks::VksChecker<E> as zksync_circuit_breaker::CircuitBreaker>::check::{{closure}}::h97a7fb2f6a56933a
zksync_1    |   17:     0x55fa24c786e1 - zksync_core::initialize_components::{{closure}}::hd2de05e0b08fa0ae
zksync_1    |   18:     0x55fa24cab7f8 - zksync_server::main::{{closure}}::h6a9062baf3475375
zksync_1    |   19:     0x55fa24ca32ae - tokio::runtime::park::CachedParkThread::block_on::hb80b3c05ba94cc2a
zksync_1    |   20:     0x55fa24e644a9 - tokio::runtime::scheduler::multi_thread::MultiThread::block_on::hdce881f9aeaa7868
zksync_1    |   21:     0x55fa24e90c0e - tokio::runtime::runtime::Runtime::block_on::h83c7f20192374e00
zksync_1    |   22:     0x55fa24ed3c3f - zksync_server::main::h0fc6d8059226339d
zksync_1    |   23:     0x55fa24d205e3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h19fee4b7580c7a4f
zksync_1    |   24:     0x55fa24e9d73d - std::rt::lang_start::{{closure}}::h9c0e3aabb848b079
zksync_1    |   25:     0x55fa26981a1c - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hfa1c3687c9a20bb8
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/ops/function.rs:606:13
zksync_1    |   26:     0x55fa26981a1c - std::panicking::try::do_call::h0497133ebe1341e5
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:483:40
zksync_1    |   27:     0x55fa26981a1c - std::panicking::try::h6c3de05c7ca5d07f
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:447:19
zksync_1    |   28:     0x55fa26981a1c - std::panic::catch_unwind::h969058ecb5334b30
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panic.rs:137:14
zksync_1    |   29:     0x55fa26981a1c - std::rt::lang_start_internal::{{closure}}::h0de6d98cafb42a58
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/rt.rs:148:48
zksync_1    |   30:     0x55fa26981a1c - std::panicking::try::do_call::h77301dcf43953993
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:483:40
zksync_1    |   31:     0x55fa26981a1c - std::panicking::try::h45b05523a2fe135a
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:447:19
zksync_1    |   32:     0x55fa26981a1c - std::panic::catch_unwind::hd95cea544b2b9bd2
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panic.rs:137:14
zksync_1    |   33:     0x55fa26981a1c - std::rt::lang_start_internal::hd16e6ff7bf05a444
zksync_1    |                                at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/rt.rs:148:20
zksync_1    |   34:     0x55fa24ed3d25 - main
zksync_1    |   35:     0x7f3dd13c709b - __libc_start_main
zksync_1    |   36:     0x55fa24c6d55a - _start
zksync_1    |   37:                0x0 - <unknown>
local-setup_zksync_1 exited with code 101`

### Have you made any recent changes to the contract before encountering this issue?

No

### Are there any external libraries or contracts that your contract interacts with?

No

### Can you provide the relevant portions of your contract code where the issue is occurring?

No

### Have you tried to isolate the problem, and if so, what were the results?

No

### What steps have you already taken to try to resolve the issue?

I tried to run with sudo permissions but not working

**Comment by dutterbutter** [comment]
Hey 👋 @seddik11 can you provide me with the steps you took to start the local node so I may attempt to reproduce as well as your environment details please :) 

To provide explicit instructions for running [`matter-labs/local-setup`](https://github.com/matter-labs/local-setup): 

1. Make sure Docker and docker-compose are installed 
2. If you have run local-setup in the past, its best to reset the zkSync state by running `./clear.sh` 
3. Run the local-setup by running `./start.sh` 

**Note:** It will take some time for contracts to be deployed so please be patient before attempting to run against test suite.

**Comment by seddik11** [comment]
Hi Thank you for your reply.
In fact I am using cloud9 and after extending the disk size everything is running well.
However I am facing gas errors that I didn't meet with Goerli ZkSync Testnet.

I am following the "Daily spending limit account" Tutorial and I am trying to deploy an Account
The first error I met is "cannot estimate gas; transaction may fail or may require manual gas limit"
so I set the gas manually like this in the transaction
` const tx = await aaFactory.deployAccount(salt, owner.address, {gasLimit: ethers.utils.parseEther("0.00000000002")});`

Now it is complaining about this "insufficient funds for intrinsic transaction cost"

BTW I tried different gasLimit values but same problem.

Thanks for help

**Reply by dutterbutter** [reply]
Hello 👋 

Thanks for the details provided. Its awesome you are following the "Daily spending limit account" tutorial!! 🚀 You mentioned that you didn't meet these issues on **Goerli zkSync Testnet**...are you trying to deploy these contracts to **zkSync Era Mainnet** now? 

Let's focus on the first error, as you shouldn't need to pass in the `gasLimit` manually in `aaFactory.deployAccount`. It is likely that the wallet being used to deploy may not have enough funds. Let's check the balance of the deployer wallet. The wallet address should be defined in your `deployFactoryAccount.ts` deployment script. You can reference the wallet address here 👉 [link](https://github.com/matter-labs/daily-spendlimit-tutorial/blob/ae011456cdf271b2b3a554956e321f98aa9ef6c6/deploy/deployFactoryAccount.ts#L9). Please ensure there is sufficient funds available in the wallet. 

Please try these steps and let me know if they help or if you encounter further issues. On a side 🎶 , what other tutorials would you like to see?

**Comment by seddik11** [comment]
Hi,
I am using a local network here is my hardhat config

`zkSyncLocal: {
        url: "http://localhost:3050",
        ethNetwork: "http://localhost:8545",
        zksync: true,
      }`

I am using one of rich wallets [here](https://github.com/matter-labs/local-setup/blob/main/rich-wallets.json)

here is the wallet balance 
`account balance is now 189738500000000` in wei

Yes It seems that the wallet balance is too low

It turns out that those rich wallets don't have enough balance and are not rich

Should i use a faucet for example ?

Thank you

**Reply by dutterbutter** [reply]
Yes, please draw testnet ETH from the faucet for your own address and thanks for letting us know!

**Reply by seddik11** [reply]
Thank you for your reactivity.
I am using [local network](https://era.zksync.io/docs/tools/hardhat/testing.html#start-the-local-nodes) not testnet
That's why using those [rich wallets](https://github.com/matter-labs/local-setup/blob/main/rich-wallets.json)

**Reply by dutterbutter** [reply]
If testing locally, those wallets should have plenty of funds. The local node deposit funds to those accounts after initialization. Feel free to post your repo and I can review.

**Reply by seddik11** [reply]
strange. because I just followed steps here [matter-labs/local-setup](https://github.com/matter-labs/local-setup) 
is it maybe because of running ./clear.sh ?

**Reply by dutterbutter** [reply, marked as answer]
No it should still re-initialize those accounts. This script checks the balances of those rich wallets connecting to a local node running on `http://localhost:3050` if you would like to check those balances. 

```
import { Provider, Wallet } from "zksync-web3";
import { HardhatRuntimeEnvironment } from "hardhat/types";
import * as ethers from "ethers";

export default async function (hre: HardhatRuntimeEnvironment) {
  console.log(`Running script...`);
  const provider = new Provider("http://localhost:3050");

  const RICH_WALLET_1 = new Wallet(
    "0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110",
  );
  const RICH_WALLET_2 = new Wallet(
    "0xac1e735be8536c6534bb4f17f06f6afc73b2b5ba84ac2cfb12f7461b20c0bbe3",
  );
  const RICH_WALLET_3 = new Wallet(
    "0xd293c684d884d56f8d6abd64fc76757d3664904e309a0645baf8522ab6366d9e",
  );
  const RICH_WALLET_4 = new Wallet(
    "0x850683b40d4a740aa6e745f889a6fdc8327be76e122f5aba645a5b02d0248db8",
  );
  const RICH_WALLET_5 = new Wallet(
    "0xf12e28c0eb1ef4ff90478f6805b68d63737b7f33abfa091601140805da450d93",
  );
  const RICH_WALLET_6 = new Wallet(
    "0xe667e57a9b8aaa6709e51ff7d093f1c5b73b63f9987e4ab4aa9a5c699e024ee8",
  );
  const RICH_WALLET_7 = new Wallet(
    "0x28a574ab2de8a00364d5dd4b07c4f2f574ef7fcc2a86a197f65abaec836d1959",
  );
  const RICH_WALLET_8 = new Wallet(
    "0x74d8b3a188f7260f67698eb44da07397a298df5427df681ef68c45b34b61f998",
  );
  const RICH_WALLET_9 = new Wallet(
    "0xbe79721778b48bcc679b78edac0ce48306a8578186ffcb9f2ee455ae6efeace1",
  );
  const RICH_WALLET_10 = new Wallet(
    "0x3eb15da85647edd9a1159a4a13b9e7c56877c4eb33f614546d4db06a51868b1c",
  );

  let rich_wallet_1_balance = await provider.getBalance(RICH_WALLET_1.address);
  let rich_wallet_2_balance = await provider.getBalance(RICH_WALLET_2.address);
  let rich_wallet_3_balance = await provider.getBalance(RICH_WALLET_3.address);
  let rich_wallet_4_balance = await provider.getBalance(RICH_WALLET_4.address);
  let rich_wallet_5_balance = await provider.getBalance(RICH_WALLET_5.address);
  let rich_wallet_6_balance = await provider.getBalance(RICH_WALLET_6.address);
  let rich_wallet_7_balance = await provider.getBalance(RICH_WALLET_7.address);
  let rich_wallet_8_balance = await provider.getBalance(RICH_WALLET_8.address);
  let rich_wallet_9_balance = await provider.getBalance(RICH_WALLET_9.address);
  let rich_wallet_10_balance = await provider.getBalance(
    RICH_WALLET_10.address,
  );

  console.log(
    `Rich wallet 1 ETH balance is now ${ethers.utils.formatEther(rich_wallet_1_balance).toString()}`,
  );
  console.log(
    `Rich wallet 2 ETH balance is now ${ethers.utils.formatEther(rich_wallet_2_balance).toString()}`,
  );
  console.log(
    `Rich wallet 3 ETH balance is now ${ethers.utils.formatEther(rich_wallet_3_balance).toString()}`,
  );
  console.log(
    `Rich wallet 4 ETH balance is now ${ethers.utils.formatEther(rich_wallet_4_balance).toString()}`,
  );
  console.log(
    `Rich wallet 5 ETH balance is now ${ethers.utils.formatEther(rich_wallet_5_balance).toString()}`,
  );
  console.log(
    `Rich wallet 6 ETH balance is now ${ethers.utils.formatEther(rich_wallet_6_balance).toString()}`,
  );
  console.log(
    `Rich wallet 7 ETH balance is now ${ethers.utils.formatEther(rich_wallet_7_balance).toString()}`,
  );
  console.log(
    `Rich wallet 8 ETH balance is now ${ethers.utils.formatEther(rich_wallet_8_balance).toString()}`,
  );
  console.log(
    `Rich wallet 9 ETH balance is now ${ethers.utils.formatEther(rich_wallet_9_balance).toString()}`,
  );
  console.log(
    `Rich wallet 10 ETH balance is now ${ethers.utils.formatEther(rich_wallet_10_balance).toString()}`,
  );

  console.log(`Done!`);
}
```

It should output the following:

```
Running script...
Rich wallet 1 ETH balance is now 1000000000000.0029834955
Rich wallet 2 ETH balance is now 1000000000000.0029868175
Rich wallet 3 ETH balance is now 1000000000000.0029868175
Rich wallet 4 ETH balance is now 1000000000000.0029868175
Rich wallet 5 ETH balance is now 1000000000000.0029868175
Rich wallet 6 ETH balance is now 1000000000000.0029868175
Rich wallet 7 ETH balance is now 1000000000000.0029868175
Rich wallet 8 ETH balance is now 1000000000000.0029868175
Rich wallet 9 ETH balance is now 1000000000000.0029868175
Rich wallet 10 ETH balance is now 1000000000000.0029868175
Done!
```

If it returns a healthy balance, then its likely you are not targeting the local node you have running but rather the testnet.

**Comment by seddik11** [comment]
Yes exactly,
In fact my profider url was pointing to testnet
I was able to deploy an account on Local net 
Thank you so much

**Comment by valensov** [comment]
Oh, thank you!

**Comment by EvieMmM** [comment]
Thx! Rly!

**Comment by Nniikka** [comment]
Goog job.
